src/Pact/Parse.hs

```diff
@@ -87,9 +87,13 @@ number = do
   let strToNum = foldl' (\x d -> 10*x + toInteger (digitToInt d))
   case dec of
     Nothing -> return $ LInteger (neg (strToNum 0 num))
-    Just d -> return $ LDecimal $ Decimal
-              (fromIntegral (length d))
-              (neg (strToNum (strToNum 0 num) d))
+    Just d ->
+      if length d > 255
+      then fail $ "decimal literal with too many digits to the right of the \
+        \decimal point (255 max): " ++ show num ++ "." ++ show d
+      else return $ LDecimal $ Decimal
```

Comment:
We've been avoiding backslash-multiline for pact code (except for natives which are necessarily verbose) and in any case this is a long error relative to others, suggest "decimal precision overflow (255 max)" or similar